src/js/pagination.js

```diff
@@ -0,0 +1,109 @@
+document.addEventListener("DOMContentLoaded", async () => {
+  setupPagination(fetchAndRenderExercises);
+});
+
+import api from "./api";
+
+async function fetchAndRenderExercises(page) {
+  const data = await getExercises(page);
+  console.log("Current Page Data:", data.results);
+
+}
+
+let firstBtn;
+let prevBtn;
+let nextBtn;
+let lastBtn;
+let pageInfo;
+
+export const getContentPagination = () => {
+  firstBtn = document.getElementById("first-btn");
+  prevBtn = document.getElementById("prev-btn");
+  nextBtn = document.getElementById("next-btn");
+  lastBtn = document.getElementById("last-btn");
+  pageInfo = document.getElementById("page-info");
+};
+
+async function setupPagination(callback) {
+  getContentPagination();
+
+
+  const initialData = await getExercises(1);
+  const totalPages = initialData.totalPages;
+  let currentPage = 1;
+
+  function renderPagination() { 
+    let pages = [];
+
+    if (totalPages <= 3) {
+      for (let i = 1; i <= totalPages; i++) {
+        pages.push(i);
+      }
+    } else {
+      if (currentPage > 2) {
+        pages.push(1);
+        pages.push("...");
+      }
+
+      let startPage = Math.max(1, currentPage - 1);
+      let endPage = Math.min(totalPages, currentPage + 1);
+
+      if (currentPage === 1) {
+        endPage = 3;
+      } else if (currentPage === totalPages) {
+        startPage = totalPages - 2;
+      }
+
+      for (let i = startPage; i <= endPage; i++) {
+        pages.push(i);
+      }
+
+      if (currentPage < totalPages - 1) {
+        pages.push("...");
+        pages.push(totalPages);
+      }
+    }
+
+    pageInfo.innerHTML = pages
+      .map(page => {
+        if (page === "...") {
+          return `<span class="dots">${page}</span>`;
+        }
+        return `<button class="page-number ${page === currentPage ? 'active' : ''}" data-page="${page}" value="${page}">${page}</button>`;
+      })
+      .join(" ");
+    
+ 
+    document.querySelectorAll(".page-number").forEach(button => {
+      button.addEventListener("click", (event) => {
+        goToPage(Number(event.target.value), callback);
+      });
+    });
+  }
+  
+  async function goToPage(page, callback) {
```

Comment:
Please move this function out of setupPagination and pass needed params, in the setupPagination only call when needed. The function is not async, we do not wait for callback call

Reply:
moved